### ISSUE TITLE
Sidebar scroll not working

### ISSUE THREAD

**Issue by BeastMaster07** [react-bootstrap/react-bootstrap#3765]
## Describe the bug

Sidebar scroll is not working properly

## To Reproduce

1. Go to react-bootstrap docs
2. View any component 
3. Scroll the particular component and then scroll the sidebar .

## Expected behavior

Scroll any component and then scroll the sidebar , it is not working most of the times. It works only on clicking the sidebar and not by hover ( which is ideal, I guess).
If the sidebar scroll works, just double click on the main component screen and now try to scroll the sidebar, it won't work.

## Environment (please complete the following information)

- Operating System: Windows 10
- Browser, Version: Chrome 74
- React-Bootstrap Version: 1.0.0-beta.6

**Comment by taion** [comment]
Ouch. It's worth noting that the same thing happens on upstream docs as well, though.

**Comment by jquense** [comment]
what the heck? this seems like a change in how the browser determines the "active" scroll area, needing to "activate" it is new and weird

**Comment by taion** [comment]
I think it might have to do with the double-sidebar layout. I don't see this on e.g. our single sidebar page.

**Comment by jquense** [comment]
I can't find an upstream issue maybe they are also unaware

**Comment by BeastMaster07** [comment]
Any updates?

**Comment by peterkmurphy** [comment]
This issue has been around for more than a year. I observe it both on Firefox 74.0 and Google Chrome 83.0 (on Linux Mint 18.3).

@taion : any idea what can be done and when it can be solved? Because this is a serious **_disincentive_** for using react-bootstrap. I don't want to be nasty, but a product in such as state that it destroys its own documentation....

**Comment by jquense** [comment]
> any idea what can be done and when it can be solved?

fastest way to get something fixed is to send a PR. 

> but a product in such as state

This isn't a product, it's an open source project built by people in their free time with free labor.

>  I don't want to be nasty

Then don't be. be productive and jump in and help

**Comment by peterkmurphy** [comment]
@jquense: I can do my little bit by observing that on the[ Components page](https://react-bootstrap.github.io/components/alerts/), I see:

```
@media (min-width:768px) {
 .SideNav-OverflowWrapper-module--cls2--2_4Kd {
  overflow:hidden;
  display:block!important;
  height:100%!important
 }
}
```
Turn that overflow to "auto", and your problem is solved. But would this one-liner fix causing problems elsewhere? 

I wouldn't know, because I don't have the (for want of a better word) _institutional_ knowledge of the react-bootstrap project, up to and including what test software you are using. Learning this stuff takes time, and time is what I don't have much of at the moment. 

With the utmost respect, I suggest that the PR comes from yourselves. You'd be able to see the consequences of this one liner fix far, far, far quicker that me.

**Comment by jquense** [comment]
We totally understand that you may not have time to work on open source. Folks here, will get to it when they have time and the energy. there are higher priority things  to do. In the meantime it not helpful nor respectful to take the time to chastise maintainers, whose work you benefit from for free.

**Comment by peterkmurphy** [comment]
@jquense: Are you actually interested in solving the bug?

**Comment by jquense** [comment]
Yes that is why the issue is still open

**Comment by peterkmurphy** [comment]
Excellent. So is the issue just a one liner fix?

**Comment by taion** [comment]
Looks like that's what upstream TWBS has, too; we'd happily take a PR to make things match.

**Comment by peterkmurphy** [comment]
Brilliant. Are there any links on setting up a test environment so as to make sure a PR would work?

**Comment by peterkmurphy** [comment]
Or just to add to my own comment: is everything I need to know [available from the home page of this repository](https://github.com/react-bootstrap/react-bootstrap/)?

**Comment by kyletsang** [comment]
Correct - see "Local setup" for instructions

**Comment by peterkmurphy** [comment]
I installed yarn, and ran the tests. All were successful. The issues came when it came to start a local copy of the docs site:

```
peter@Peter-PC ~/projects/react-bootstrap $ yarn start
yarn run v1.22.4
$ yarn --cwd www develop
$ gatsby develop
/bin/sh: 1: gatsby: not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Since I don't want to go down a blind tunnel, is the "docs site" the same site as the one with the buggy sidebar scroll that is the subject of this bug report? If so, is there any special instructions for installing gatsby I should be aware of?

**Comment by taion** [comment]
You need to run `yarn` (install) in the `www/` subdirectory. Or run `yarn bootstrap` at root.

**Comment by peterkmurphy** [comment]
Thank you, @taion, that helped. I ran yarn in the www/ subdirectory as you suggested.

After some searching, I found this code in www/src/components/SideNav.js:

```
const OverflowWrapper = styled('div')`
  @import '../css/theme';

  @include media-breakpoint-up(md) {
    overflow: hidden;
    display: block !important;
    height: 100% !important;
  }
`;
```

If I remove the "overflow: hidden;" line, I see the sidebar scroll bar, which is what I want. But the fix can't be that simple, can it? Is there some purpose to OverflowWrapper that I should be aware of?

**Comment by taion** [comment]
i don't think that quite puts the scrollbar in the right place – compare where it ends up v. where it shows up on the upstream TWBS site

**Comment by peterkmurphy** [comment]
I'm sorry, but I don't understand "TWBS". Do you have an URL that I can use as a basis for comparison?

**Comment by taion** [comment]
https://getbootstrap.com/docs/4.5/components/alerts/

**Comment by peterkmurphy** [comment]
Thank you.

**Comment by peterkmurphy** [comment]
In the same SideNav.js file, I changed the margin-right and padding-right values in the TableOfContents component as so. The scrollbar seems to be the right place on my machine. Is there any issues with this provisional solution on other people's computers?   

```
const TableOfContents = styled('nav')`
  @import '../css/theme';

  composes: pt-2 pb-4 from global;

  @include media-breakpoint-up(md) {
    height: 100% !important;
    overflow: auto;
    margin-right: -15px;
    padding-right: calc(15px + 1rem);
  }
`;
```

**Comment by taion** [comment]
Do you want to just throw up a PR? Our build process actually builds the docs page and puts it up on Netlify, so we can see it there.

**Comment by jquense** [comment]
we could probably just style the scrollbar and avoid the fancy workaround here ?

**Comment by peterkmurphy** [comment]
Here's the pull request.

https://github.com/react-bootstrap/react-bootstrap/pull/5358
